File: typescript/src/resources/usage.ts
import type { UsageResponse, UsageBalance } from '../types/usage.js';

/**
 * Usage resource — token consumption and credit tracking.
 *
 * Endpoints: GET /usage, GET /usage/balance
 */
export interface UsageResource {
  /** Get token consumption and cost breakdown. */
  get(): Promise<UsageResponse>;

  /** Get token credit balance. */
  getBalance(): Promise<UsageBalance>;
}

File: python/copass-core/src/copass_core/resources/usage.py
"""Usage resource — token consumption and credit tracking.

Port of ``typescript/packages/core/src/resources/usage.ts``.
"""

from __future__ import annotations

from typing import Any, Dict

from copass_core.resources.base import BaseResource


class UsageResource(BaseResource):
    async def get_summary(self) -> Dict[str, Any]:
        return await self._get("/api/v1/usage")

    async def get_balance(self) -> Dict[str, Any]:
        return await self._get("/api/v1/usage/credits")


__all__ = ["UsageResource"]
